File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkTestAudioDeviceHelper.m
#import "FlutterZoomVideoSdkTestAudioDeviceHelper.h"
#import "JSONConvert.h"

@implementation FlutterZoomVideoSdkTestAudioDeviceHelper

- (ZoomVideoSDKTestAudioDeviceHelper *)getTestAudioDeviceHelper {
    ZoomVideoSDKTestAudioDeviceHelper* testAudioDeviceHelper = nil;
    @try {
        testAudioDeviceHelper = [[ZoomVideoSDK shareInstance] getTestAudioDeviceHelper];
        if (testAudioDeviceHelper == nil) {
            NSException *e = [NSException exceptionWithName:@"NoTestAudioDeviceHelperFound" reason:@"No Test Audio Device Helper Found" userInfo:nil];
            @throw e;
        }
    } @catch(NSException *e) {
        NSLog(@"%@ - %@", e.name, e.reason);
    }
    return testAudioDeviceHelper;
}

-(void) startMicTest: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getTestAudioDeviceHelper] startMicTest])]);
    });
}

-(void) stopMicTest: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getTestAudioDeviceHelper] stopMicTest])]);
   });
}

-(void) playMicTest: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getTestAudioDeviceHelper] playMicTest])]);
   });
}

-(void) startSpeakerTest: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getTestAudioDeviceHelper] startSpeakerTest])]);
   });
}

-(void) stopSpeakerTest: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getTestAudioDeviceHelper] stopSpeakerTest])]);
   });
}

@end

File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkTestAudioDeviceHelper.java
package com.flutterzoom.videosdk;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkErrors;

import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDK;
import us.zoom.sdk.ZoomVideoSDKTestAudioDeviceHelper;

public class FlutterZoomVideoSdkTestAudioDeviceHelper {

    private Activity activity;

    FlutterZoomVideoSdkTestAudioDeviceHelper(Activity activity) {
        this.activity = activity;
    }

    private ZoomVideoSDKTestAudioDeviceHelper getTestAudioDeviceHelper() {
        ZoomVideoSDKTestAudioDeviceHelper testAudioDeviceHelper = null;
        try {
            testAudioDeviceHelper = ZoomVideoSDK.getInstance().getTestAudioDeviceHelper();
            if (testAudioDeviceHelper == null) {
                throw new Exception ("Test audio is not available");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return testAudioDeviceHelper;
    }

    public void startMicTest(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                getTestAudioDeviceHelper().stopMicTest();
                result.success(FlutterZoomVideoSdkErrors.valueOf(getTestAudioDeviceHelper().startMicTest()));
            }
        });
    }

    public void stopMicTest(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getTestAudioDeviceHelper().stopMicTest()));
            }
        });
    }

    public void playMicTest(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getTestAudioDeviceHelper().playMicTest()));
            }
        });
    }

    public void startSpeakerTest(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getTestAudioDeviceHelper().startSpeakerTest()));
            }
        });
    }

    public void stopSpeakerTest(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getTestAudioDeviceHelper().stopSpeakerTest()));
            }
        });
    }

}
